<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>716</width>
    <height>379</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>基础控件测试</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="page_0">
       <layout class="QGridLayout" name="gridLayout_15">
        <item row="0" column="0">
         <widget class="QGroupBox" name="gBox_datetime">
          <property name="title">
           <string>Date and Time</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="0">
            <widget class="QTimeEdit" name="timeEdit"/>
           </item>
           <item row="1" column="0">
            <widget class="QDateEdit" name="dateEdit"/>
           </item>
           <item row="2" column="0">
            <widget class="QDateTimeEdit" name="dateTimeEdit"/>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QCalendarWidget" name="calendarWidget"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_1">
       <layout class="QGridLayout" name="gridLayout_13">
        <item row="0" column="0" rowspan="5">
         <widget class="QToolBox" name="toolBox">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="page">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>337</width>
             <height>215</height>
            </rect>
           </property>
           <attribute name="label">
            <string>Page 1</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_9">
            <item row="0" column="0">
             <widget class="QCommandLinkButton" name="commandLinkButton">
              <property name="text">
               <string>CommandLinkButton</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_2">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>337</width>
             <height>215</height>
            </rect>
           </property>
           <attribute name="label">
            <string>Page 2</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_16">
            <item row="0" column="0">
             <widget class="QKeySequenceEdit" name="keySequenceEdit"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QDialogButtonBox" name="buttonBox">
          <property name="standardButtons">
           <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="1" rowspan="3">
         <widget class="QGroupBox" name="groupBox_7">
          <property name="title">
           <string>GroupBox</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_12">
           <item row="0" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QDial" name="dial"/>
             </item>
             <item>
              <widget class="QLCDNumber" name="lcdNumber">
               <property name="digitCount">
                <number>3</number>
               </property>
               <property name="intValue" stdset="0">
                <number>0</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_2">
       <layout class="QGridLayout" name="gridLayout_10">
        <item row="0" column="0">
         <widget class="QGroupBox" name="gBox_btn">
          <property name="title">
           <string>QPushButton</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_8">
           <item row="0" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QPushButton" name="btn_1">
               <property name="text">
                <string>按钮1</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_2">
               <property name="text">
                <string>按钮2</string>
               </property>
               <property name="iconSize">
                <size>
                 <width>16</width>
                 <height>16</height>
                </size>
               </property>
               <property name="checkable">
                <bool>false</bool>
               </property>
               <property name="autoDefault">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QPushButton" name="btn_3">
               <property name="text">
                <string>按钮3</string>
               </property>
               <property name="autoRepeat">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_4">
               <property name="text">
                <string>按钮4</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1" colspan="2">
         <widget class="QGroupBox" name="gBox_Box">
          <property name="title">
           <string>ComboBox and SpinBox</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="2" column="3">
            <widget class="QComboBox" name="comboBox"/>
           </item>
           <item row="2" column="0">
            <widget class="QSpinBox" name="spinBox_R"/>
           </item>
           <item row="2" column="2">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_B"/>
           </item>
           <item row="1" column="0" colspan="4">
            <widget class="QFontComboBox" name="fontComboBox"/>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="spinBox_G"/>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QGroupBox" name="gBox_tbtn">
          <property name="title">
           <string>QToolButton</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_11">
           <item row="0" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QToolButton" name="tbtn_1">
               <property name="minimumSize">
                <size>
                 <width>50</width>
                 <height>50</height>
                </size>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="tbtn_2">
               <property name="minimumSize">
                <size>
                 <width>50</width>
                 <height>50</height>
                </size>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="iconSize">
                <size>
                 <width>10</width>
                 <height>10</height>
                </size>
               </property>
               <property name="checkable">
                <bool>false</bool>
               </property>
               <property name="autoRaise">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_9">
             <item>
              <widget class="QToolButton" name="tbtn_3">
               <property name="minimumSize">
                <size>
                 <width>50</width>
                 <height>50</height>
                </size>
               </property>
               <property name="focusPolicy">
                <enum>Qt::NoFocus</enum>
               </property>
               <property name="autoFillBackground">
                <bool>false</bool>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="checkable">
                <bool>false</bool>
               </property>
               <property name="autoRepeat">
                <bool>false</bool>
               </property>
               <property name="autoExclusive">
                <bool>false</bool>
               </property>
               <property name="autoRaise">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="tbtn_4">
               <property name="minimumSize">
                <size>
                 <width>50</width>
                 <height>50</height>
                </size>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QGroupBox" name="gBox_rbtn">
          <property name="title">
           <string>QRadioButton</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QRadioButton" name="rBtn_1">
               <property name="text">
                <string>RadioButton</string>
               </property>
               <property name="autoRepeat">
                <bool>false</bool>
               </property>
               <property name="autoExclusive">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="rBtn_3">
               <property name="text">
                <string>RadioButton</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="0" column="1">
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <widget class="QRadioButton" name="rBtn_2">
               <property name="text">
                <string>RadioButton</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="rBtn_4">
               <property name="text">
                <string>RadioButton</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QGroupBox" name="gBox_cBox">
          <property name="title">
           <string>QCheckBox</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="0" column="0">
            <layout class="QVBoxLayout" name="verticalLayout_6">
             <item>
              <widget class="QCheckBox" name="cBox_1">
               <property name="text">
                <string>CheckBox</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="cBox_3">
               <property name="text">
                <string>CheckBox</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="0" column="1">
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <widget class="QCheckBox" name="cBox_2">
               <property name="text">
                <string>CheckBox</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="cBox_4">
               <property name="text">
                <string>CheckBox</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_3">
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <widget class="QScrollArea" name="scrollArea">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>-82</y>
             <width>664</width>
             <height>357</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <widget class="QGroupBox" name="gBox_lab">
              <property name="title">
               <string>QLabel</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_7">
               <item>
                <widget class="QLabel" name="lab_1">
                 <property name="text">
                  <string>显示文字</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="lab_2">
                 <property name="text">
                  <string>显示图片</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="lab_3">
                 <property name="text">
                  <string>显示动画</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="lab_4">
                 <property name="text">
                  <string>显示HTMl</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QGroupBox" name="gBox_lineedit">
              <property name="title">
               <string>QLineEdit</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <widget class="QLineEdit" name="lineEdit_1">
                 <property name="text">
                  <string>只读文本框</string>
                 </property>
                 <property name="readOnly">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="lineEdit_2">
                 <property name="placeholderText">
                  <string>默认显示</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="lineEdit_3">
                 <property name="text">
                  <string/>
                 </property>
                 <property name="placeholderText">
                  <string>正则表达式</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="lineEdit_4">
                 <property name="text">
                  <string/>
                 </property>
                 <property name="echoMode">
                  <enum>QLineEdit::Password</enum>
                 </property>
                 <property name="readOnly">
                  <bool>false</bool>
                 </property>
                 <property name="placeholderText">
                  <string>密码</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QGroupBox" name="gBox_textedit">
              <property name="title">
               <string>TextEdit</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_3">
                 <item>
                  <widget class="QTextBrowser" name="textBrowser"/>
                 </item>
                 <item>
                  <widget class="QPlainTextEdit" name="plainTextEdit"/>
                 </item>
                </layout>
               </item>
               <item>
                <widget class="QTextEdit" name="textEdit"/>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_4">
       <layout class="QGridLayout" name="gridLayout_14">
        <item row="0" column="0">
         <widget class="QGroupBox" name="gBox_bar">
          <property name="title">
           <string>ProgressBar  Slider   ScrollBar</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QProgressBar" name="progressBar">
               <property name="value">
                <number>24</number>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="textDirection">
                <enum>QProgressBar::TopToBottom</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="horizontalSlider">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QScrollBar" name="horizontalScrollBar">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="0" column="1">
            <widget class="QSlider" name="verticalSlider">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QScrollBar" name="verticalScrollBar">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>716</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_page">
    <property name="title">
     <string>基础控件</string>
    </property>
    <addaction name="action_btn"/>
    <addaction name="action_text"/>
    <addaction name="action_bar"/>
    <addaction name="action_datetime"/>
    <addaction name="action_etc"/>
   </widget>
   <widget class="QMenu" name="menu_style">
    <property name="title">
     <string>样式表</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>关于</string>
    </property>
    <addaction name="action_about"/>
    <addaction name="action_exit"/>
   </widget>
   <addaction name="menu_page"/>
   <addaction name="menu_style"/>
   <addaction name="menu"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_btn">
   <property name="text">
    <string>按钮</string>
   </property>
  </action>
  <action name="action_text">
   <property name="text">
    <string>文本</string>
   </property>
  </action>
  <action name="action_bar">
   <property name="text">
    <string>进度条</string>
   </property>
  </action>
  <action name="action_datetime">
   <property name="text">
    <string>时间日期</string>
   </property>
  </action>
  <action name="action_etc">
   <property name="text">
    <string>其他</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="text">
    <string>退出</string>
   </property>
  </action>
  <action name="action_about">
   <property name="text">
    <string>关于Qt</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>verticalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>progressBar</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>665</x>
     <y>288</y>
    </hint>
    <hint type="destinationlabel">
     <x>603</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>progressBar</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>177</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dial</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>lcdNumber</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>471</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>611</x>
     <y>163</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>progressBar</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>315</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>152</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>progressBar</sender>
   <signal>valueChanged(int)</signal>
   <receiver>verticalSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>637</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>665</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
